request.go

```diff
@@ -62,74 +62,88 @@ func doRequest(ctx context.Context, url string, m *dns.Msg) (*dns.Msg, error) {
 	return r, nil
 }
 
-func doRequestA(ctx context.Context, url string, domain string) ([]net.IPAddr, error) {
+func doRequestA(ctx context.Context, url string, domain string) ([]net.IPAddr, uint32, error) {
 	fqdn := dns.Fqdn(domain)
 
 	m := new(dns.Msg)
 	m.SetQuestion(fqdn, dns.TypeA)
 
 	r, err := doRequest(ctx, url, m)
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
+	var ttl uint32
 	result := make([]net.IPAddr, 0, len(r.Answer))
 	for _, rr := range r.Answer {
 		switch v := rr.(type) {
 		case *dns.A:
 			result = append(result, net.IPAddr{IP: v.A})
+			if ttl == 0 || v.Hdr.Ttl < ttl {
+				ttl = v.Hdr.Ttl
+			}
 		default:
 			log.Warnf("unexpected DNS resource record %+v", rr)
 		}
 	}
 
-	return result, nil
+	return result, ttl, nil
 }
 
-func doRequestAAAA(ctx context.Context, url string, domain string) ([]net.IPAddr, error) {
+func doRequestAAAA(ctx context.Context, url string, domain string) ([]net.IPAddr, uint32, error) {
 	fqdn := dns.Fqdn(domain)
 
 	m := new(dns.Msg)
 	m.SetQuestion(fqdn, dns.TypeAAAA)
 
 	r, err := doRequest(ctx, url, m)
 	if err != nil {
-		return nil, err
+		return nil, 0, err
 	}
 
+	var ttl uint32
 	result := make([]net.IPAddr, 0, len(r.Answer))
 	for _, rr := range r.Answer {
 		switch v := rr.(type) {
 		case *dns.AAAA:
 			result = append(result, net.IPAddr{IP: v.AAAA})
+			if ttl == 0 || v.Hdr.Ttl < ttl {
+				ttl = v.Hdr.Ttl
+			}
+
 		default:
 			log.Warnf("unexpected DNS resource record %+v", rr)
 		}
 	}
 
-	return result, nil
+	return result, ttl, nil
 }
 
-func doRequestTXT(ctx context.Context, url string, domain string) ([]string, error) {
+func doRequestTXT(ctx context.Context, url string, domain string) ([]string, uint32, error) {
```

Comment:
Is there a reason to use a `uint32` here, and not a `time.Duration`?

Reply:
that's what the DNS library gives us, and the actual standard TTL type in the DNS spec.